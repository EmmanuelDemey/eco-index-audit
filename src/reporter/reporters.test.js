const csv = require("./csv");
const json = require("./json");

const sampleAudit = {
  ecoIndex: 56,
  grade: "C",
  greenhouseGasesEmission: 1.88,
  waterConsumption: 2.82,
  pages: [
    {
      url: "url",
      ecoIndex: 56,
      grade: "C",
      greenhouseGasesEmission: 1.88,
      waterConsumption: 2.82,
      metrics: [
        {
          name: "number_requests",
          value: 16,
          status: "info",
          recommandation: "< 30 requests",
        },
        {
          name: "page_size_kbytes",
          value: 1200,
          status: "warning",
          recommandation: "< 1000kb",
        },
        {
          name: "Page_complexity",
          value: 3000,
          status: "error",
          recommandation: "Between 300 and 500 nodes",
        },
      ],
    },
  ],
};

test("should report with CSV", () => {
  const consoleLogMock = jest.spyOn(global.console, "log").mockImplementation();
  csv(sampleAudit, { visits: 2000 });
  const result = consoleLogMock.mock.calls.join("\n");
  console.log(result)
  expect(result).toEqual(`Métrique,Valeur,Informations complémentaires
EcoIndex,56/100,
Note,C,
GES,1.88eqCO2,Pour un total de 2000 visites par mois, ceci correspond à 42km en voiture (Peugeot 208 5P 1.6 BlueHDi FAP (75ch) BVM5)
Eau,2.82cl,Pour un total de 2000 visites par mois, ceci correspond à 1 douche`);
  consoleLogMock.mockRestore();
});

test("should report with JSON", () => {
  const expected = JSON.stringify({
    score: 56,
    grade: "C",
    estimatation_co2: {
      comment:
        "Pour un total de 2000 visites par mois, ceci correspond à 42km en voiture (Peugeot 208 5P 1.6 BlueHDi FAP (75ch) BVM5)",
      commentDetails: { numberOfVisit: 2000, value_km: 42, value: 42, unit: 'km' },
    },
    estimatation_water: {
      comment: "Pour un total de 2000 visites par mois, ceci correspond à 1 douche",
      commentDetails: { numberOfVisit: 2000, value_shower: 1, value: 1, unit: 'douches' },
    },
    pages: [
      {
        url: "url",
        ecoIndex: 56,
        grade: "C",
        greenhouseGasesEmission: 1.88,
        waterConsumption: 2.82,
        metrics: [
          {
            name: "number_requests",
            value: 16,
            status: "info",
            recommandation: "< 30 requests",
          },
          {
            name: "page_size_kbytes",
            value: 1200,
            status: "warning",
            recommandation: "< 1000kb",
          },
          {
            name: "Page_complexity",
            value: 3000,
            status: "error",
            recommandation: "Between 300 and 500 nodes",
          },
        ],
        estimatation_co2: {
          comment:
            "Pour un total de 2000 visites par mois, ceci correspond à 42km en voiture (Peugeot 208 5P 1.6 BlueHDi FAP (75ch) BVM5)",
          commentDetails: { numberOfVisit: 2000, value_km: 42, value: 42, unit: 'km' },
        },
        estimatation_water: {
          comment: "Pour un total de 2000 visites par mois, ceci correspond à 1 douche",
          commentDetails: { numberOfVisit: 2000, value_shower: 1, value: 1, unit: 'douches' },
        }
      },
    ],
  });

  const consoleLogMock = jest.spyOn(global.console, "log").mockImplementation();
  json(sampleAudit, { visits: 2000 });
  const result = consoleLogMock.mock.calls.join("\n");
  expect(JSON.stringify(JSON.parse(result))).toEqual(expected);
  consoleLogMock.mockRestore();
});
