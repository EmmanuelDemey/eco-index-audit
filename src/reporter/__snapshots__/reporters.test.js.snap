// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should report with CSV 1`] = `
"Métrique,Valeur,Informations complémentaires
EcoIndex,56/100,
Note,C,
GES,1.88eqCO2,Pour un total de 2000 visites par mois, ceci correspond à 42km en voiture (Peugeot 208 5P 1.6 BlueHDi FAP (75ch) BVM5)
Eau,2.82cl,Pour un total de 2000 visites par mois, ceci correspond à 1 douche"
`;

exports[`should report with JSON 1`] = `
"[
  {
    "label": "EcoIndex",
    "value": 56
  },
  {
    "label": "Note",
    "value": "C"
  },
  {
    "label": "GES",
    "value": 1.88,
    "unit": "gCO2e",
    "comment": "Pour un total de 2000 visites par mois, ceci correspond à 42km en voiture (Peugeot 208 5P 1.6 BlueHDi FAP (75ch) BVM5)"
  },
  {
    "label": "Eau",
    "value": 2.82,
    "unit": "cl",
    "comment": "Pour un total de 2000 visites par mois, ceci correspond à 1 douche"
  }
]"
`;

exports[`should report with TABLE 1`] = `
"[90m┌──────────┬───────────┬────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────┐[39m
[90m│[39m[31m Métrique [39m[90m│[39m[31m Valeur    [39m[90m│[39m[31m Informations complémentaires                                                                                           [39m[90m│[39m
[90m├──────────┼───────────┼────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────┤[39m
[90m│[39m EcoIndex [90m│[39m 56/100    [90m│[39m                                                                                                                        [90m│[39m
[90m├──────────┼───────────┼────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────┤[39m
[90m│[39m Note     [90m│[39m C         [90m│[39m                                                                                                                        [90m│[39m
[90m├──────────┼───────────┼────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────┤[39m
[90m│[39m GES      [90m│[39m 1.88eqCO2 [90m│[39m Pour un total de 2000 visites par mois, ceci correspond à 42km en voiture (Peugeot 208 5P 1.6 BlueHDi FAP (75ch) BVM5) [90m│[39m
[90m├──────────┼───────────┼────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────┤[39m
[90m│[39m Eau      [90m│[39m 2.82cl    [90m│[39m Pour un total de 2000 visites par mois, ceci correspond à 1 douche                                                     [90m│[39m
[90m└──────────┴───────────┴────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────┘[39m"
`;
